sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBuffer.scala

```diff
@@ -72,6 +72,11 @@ trait RapidsBuffer extends AutoCloseable {
   /** The storage tier for this buffer */
   val storageTier: StorageTier
 
+  /**
+   * A callback function that can be used for metrics when a buffer is spilled
+   */
+  val spillCallback: (String, Long) => Unit
```

Comment:
I started out with not putting defaults everywhere because I wanted to be sure that I had plumbed it through properly in all places with compile errors.  If you want me to move it to RapidsBufferId I can do it.